lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -377,18 +382,24 @@ def add_fraud_fields(post, options)
       def add_external_cardholder_authentication_data(post, options)
         return unless threeds_2_options = options[:three_d_secure]
 
-        authentication_data = {}
-        authentication_data[:acsTransactionId] = threeds_2_options[:acs_transaction_id] if threeds_2_options[:acs_transaction_id]
-        authentication_data[:cavv] = threeds_2_options[:cavv] if threeds_2_options[:cavv]
-        authentication_data[:cavvAlgorithm] = threeds_2_options[:cavv_algorithm] if threeds_2_options[:cavv_algorithm]
-        authentication_data[:directoryServerTransactionId] = threeds_2_options[:ds_transaction_id] if threeds_2_options[:ds_transaction_id]
-        authentication_data[:eci] = threeds_2_options[:eci] if threeds_2_options[:eci]
-        authentication_data[:threeDSecureVersion] = threeds_2_options[:version] if threeds_2_options[:version]
-        authentication_data[:validationResult] = threeds_2_options[:authentication_response_status] if threeds_2_options[:authentication_response_status]
-        authentication_data[:xid] = threeds_2_options[:xid] if threeds_2_options[:xid]
+        authentication_data = {
+          priorThreeDSecureData: { acsTransactionId: threeds_2_options[:acs_transaction_id] }.compact,
+          cavv: threeds_2_options[:cavv],
+          cavvAlgorithm: threeds_2_options[:cavv_algorithm],
+          directoryServerTransactionId: threeds_2_options[:ds_transaction_id],
+          eci: threeds_2_options[:eci],
+          threeDSecureVersion: threeds_2_options[:version] || options[:three_ds_version],
+          validationResult: threeds_2_options[:authentication_response_status],
+          xid: threeds_2_options[:xid],
+          acsTransactionId: threeds_2_options[:acs_transaction_id],
+          flow: threeds_2_options[:flow]
+        }.compact
 
         post['cardPaymentMethodSpecificInput'] ||= {}
         post['cardPaymentMethodSpecificInput']['threeDSecure'] ||= {}
+        post['cardPaymentMethodSpecificInput']['threeDSecure']['merchantFraudRate'] = threeds_2_options[:merchant_fraud_rate] || 99
+        post['cardPaymentMethodSpecificInput']['threeDSecure']['exemptionRequest'] = 'none'
+        post['cardPaymentMethodSpecificInput']['threeDSecure']['secureCorporatePayment'] = threeds_2_options[:secure_corporate_payment] || true
```

Comment:
Why do we set a default for these three? Does Ogone support the MOTO exemption for 3DS2?